src/streamlink/plugins/ceskatelevize.py

```diff
@@ -5,131 +5,178 @@
+            str,
+            validate.parse_json(),
+            {
+                "items": [{
+                    "items": [{
+                        validate.optional("video"): {
+                            "data": {
+                                "source": {
+                                    "playlist": [{
+                                        "id": int,
+                                        "key": str,
+                                        "date": str,
+                                    }],
+                                },
+                            },
+                        },
+                    }],
+                }],
+            },
+            validate.get(("items", 0, "items", 0, "video", "data", "source", "playlist", 0)),
+            validate.union_get("id", "key", "date"),
+        ).validate(content)
+
+        self.id = video_id
+        self.title = "ČT sport"
 
-        url = self.session.http.get(
-            data["url"],
+        return self.session.http.post(
+            "https://playlist.ceskatelevize.cz/",
+            data={
+                "data": json.dumps(
+                    {
+                        "contentType": "live",
+                        "items": [{
+                            "id": video_id,
+                            "key": f"{key}",
+                            "assetId": "CT4DRM",
+                            "playerType": "dash",
+                            "date": f"{date}",
+                            "requestSource": "front-sport",
+                            "drm": 0,
+                            "quality": "web",
+                        }],
+                    },
+                    separators=(',', ':'),
+                ),
+            },
             schema=validate.Schema(
                 validate.parse_json(),
                 {
-                    "playlist": [{
-                        validate.optional("type"): str,
-                        "streamUrls": {
-                            "main": validate.url(),
-                        }
-                    }]
+                    "RESULT": self.schema_playlist,
                 },
-                validate.get(("playlist", 0, "streamUrls", "main"))
-            )
+                validate.get(("RESULT", "playlist", 0, "streamUrls", "main")),
+            ),
         )
-        return DASHStream.parse_manifest(self.session, url)
+
+    def get_channel(self, content):
+        data = validate.Schema(
+            validate.parse_html(),
+            validate.xml_xpath_string(".//script[@id='__NEXT_DATA__'][text()]/text()"),
+            str,
+            validate.parse_json(),
+            {
+                "props": {
+                    "pageProps": {
+                        validate.optional("data"): {
+                            "liveBroadcast": {
+                                "id": str,
+                                "current": validate.none_or_all(
```

Comment:
https://www.ceskatelevize.cz/zive/ct28/

might be None
the stream is in `next`

```python
{
  'id': '28', 
  'current': None,
  'next': {'id': '28-1673778600', 'channelName': 'ČT sport Plus', 'isPlayable': True}
}
```

```python
                                "next": validate.none_or_all({
                                    "id": str,
                                    "channelName": str,
                                    "isPlayable": bool,
                                }),
```

```python
data = data.get("current") or data.get("next")
```

Reply:
OK, thanks.  I did seem to find that CT28 and others worked when available (as noted in https://github.com/streamlink/streamlink/pull/5063#issuecomment-1382009964), but I'll put this in.

Reply:
I guess this isn't in fact just for CT28, but any channel where `current` may be empty.  It wasn't something I'd seen, so I don't know when this might happen, but clearly it does.  Cheers.